backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/impl/BPartnerDAO.java

```diff
@@ -1711,4 +1712,13 @@ private static BPartnerPrintFormat toBPartnerPrintFormat(final I_C_BP_PrintForma
 				.printFormatId(PrintFormatId.ofRepoId(record.getAD_PrintFormat_ID()))
 				.build();
 	}
+
+	@Override
+	public void validateSalesRep(@NonNull final BPartnerId bPartnerId, @Nullable final BPartnerId salesRepId)
```

Comment:
this method doesn't access any data, so i think it's better up in BPartnerBL

Reply:
I wrote it like that, but BPartnerBL does not have a no-args constructor for it. Had a lot of problems writing unit-tests for it and sort of gave up. since it's not that big of a logic, it can also be in a Utility class. But could not find a good match for it.

Reply:
but you have `BPartnerSalesRepTest` in this very PR, where you actually work with BPartnerBL. 
So it looks like you mastered this art after all.
A weel thought-through utility class would be still better, but for starts i think you can just move the code..